Sources/SwiftIfConfig/StaticBuildConfiguration.swift

```diff
@@ -0,0 +1,431 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// A statically-determined build configuration that can be used with any
+/// API that requires a build configuration. Static build configurations can
+/// be (de-)serialized via Codable.
+public struct StaticBuildConfiguration: Codable {
+  public init(
+    customConditions: Set<String> = [],
+    features: Set<String> = [],
+    attributes: Set<String> = [],
+    importedModules: [String: [VersionedImportModule]] = [:],
+    targetOSNames: Set<String> = [],
+    targetArchitectures: Set<String> = [],
+    targetEnvironments: Set<String> = [],
+    targetRuntimes: Set<String> = [],
+    targetPointerAuthenticationSchemes: Set<String> = [],
+    targetObjectFileFormats: Set<String> = [],
+    targetPointerBitWidth: Int = 64,
+    targetAtomicBitWidths: [Int] = [],
+    endianness: Endianness = .little,
+    languageVersion: VersionTuple = VersionTuple(6),
+    compilerVersion: VersionTuple = VersionTuple(6, 2)
```

Comment:
I'm not super keen on having more places that we need to update version, does this need to be defaulted here?

Reply:
I suppose we don't have to default it here, no

Reply:
No, we can do without